examples/recon_example.py

```diff
@@ -0,0 +1,123 @@
+#%%
+from cil.utilities import dataexample
+from cil.utilities.display import show2D
+from cil.recon import FDK
+from cil.processors import TransmissionAbsorptionConverter, Slicer
+
+ground_truth = dataexample.SIMULATED_SPHERE_VOLUME.get()
+
+data = dataexample.SIMULATED_CONE_BEAM_DATA.get()
+twoD = True
+if twoD:
+    data = data.get_slice(vertical='centre')
+    ground_truth = ground_truth.get_slice(vertical='centre')
+
+absorption = TransmissionAbsorptionConverter()(data)
+absorption = Slicer(roi={'angle':(0, -1, 5)})(absorption)
+
+ig = ground_truth.geometry
+
+#%%
+recon = FDK(absorption, image_geometry=ig).run()
+#%%
+show2D([ground_truth, recon], title = ['Ground Truth', 'FDK Reconstruction'], origin = 'upper', num_cols = 2)
+
+# %%
+from cil.plugins.tigre import ProjectionOperator
+from cil.optimisation.algorithms import FISTA 
+from cil.optimisation.functions import LeastSquares, IndicatorBox, ZeroFunction, TotalVariation
+from cil.optimisation.operators import GradientOperator
+from cil.optimisation.utilities import callbacks
+
+#%%
+A = ProjectionOperator(image_geometry=ig, 
+                       acquisition_geometry=absorption.geometry)
+
+F = LeastSquares(A = A, b = absorption)
+G = IndicatorBox(lower=0)
+
+grad = GradientOperator(domain_geometry=ig)
+#%%
+alpha = 0.1 * A.norm()/grad.norm()
+G = alpha * TotalVariation(max_iteration=2)
+
+algo = FISTA(initial = ig.allocate(), f = F, g = G)
+
+#%%
+# Image metrics callback
+import img_quality_cil_stir as imgq
+import tensorboardX
+from datetime import datetime
+import numpy as np
+# create a tensorboardX summary writer
+dt_string = datetime.now().strftime("%Y%m%d-%H%M%S")
+tb_summary_writer = tensorboardX.SummaryWriter(f'recons/exp-{dt_string}')
+
+#%% create masks
+top = ig.allocate(0)
+bottom = ig.allocate(0)
+
+top.fill(
+    np.asarray(ground_truth.array > 0.8 * ground_truth.max(), 
+               dtype=np.float32)
+    )
+bottom.fill(
+    np.asarray(np.invert(ground_truth.array < 0.4 * ground_truth.max()), 
+               dtype=np.float32)
+)
+#%%
+show2D([ground_truth, top, bottom], num_cols=3)
+#%%
+# setup a 2 binary ROI images
+roi_image_dict = {
+    'top' : top,
+    'bottom' : bottom
+}
+
+from cil.utilities.quality_measures import mae, mse, psnr
+
+def MSE(x,y):
+    """ mean squared error between two numpy arrays
+    """
+    return ((x-y)**2).mean()
```

Comment:
MSE is generally a horrible metric as it gives too much weight to background.
Would strongly recommend NRMSE instead:

```suggestion
def NRMSE(x, y):
    """normalised mean squared error between two arrays"""
    return (((x-y)**2).mean() / (y**2).mean()) ** 0.5
```

Reply:
we can have both. For PETRIC, we want to use RMSE over (targeted) VOI (normalised by mean of background VOI

Reply:
ah if it's over VOI then sure...

Reply:
```suggestion
def MSE(x, y):
    """mean squared error between two arrays"""
    return ((x-y)**2).mean()

def NRMSE(x, y):
    """normalised mean squared error between two arrays"""
    return (((x-y)**2).mean() / (y**2).mean())**0.5
```